docs/csharp/tutorials/working-with-linq.md

```diff
@@ -80,64 +77,97 @@ static IEnumerable<string> Ranks()
 
-This method declaration also follows a standard idiom where the input and output types are `IEnumerable<T>`. That practice enables LINQ methods to be chained together to perform more complex queries.
+Let's give our extension methods a home: add a new *static* class file to your program called `Extensions.cs`, and then start building out our first extension method: 
 
 ```csharp
+// Extensions.cs
+using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace LinqFaroShuffle
 {
     public static class Extensions
     {
-        public static IEnumerable<T> InterleaveSequenceWith<T>
-            (this IEnumerable<T> first, IEnumerable<T> second)
+        public static IEnumerable<T> InterleaveSequenceWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
         {
-            // implementation coming.
+            // Our implementation will go here soon enough
         }
     }
 }
 ```
 
-You will be enumerating both sequences at once, interleaving the elements, and creating one object.  Writing a LINQ method that works with two sequences requires that you understand how `IEnumerable` works.
+Look at the method signature for a moment, specifically the parameters:
+
+```csharp
+public static IEnumerable<T> InterleaveSequenceWith<T>
+    (this IEnumerable<T> first, IEnumerable<T> second)
+```
+
+You can see the addition of the `this` modifier on the first argument to the method. That means you call the method as though it were a member method of the type of the first argument. This method declaration also follows a standard idiom where the input and output types are `IEnumerable<T>`. That practice enables LINQ methods to be chained together to perform more complex queries.
 
-The `IEnumerable` interface has one method: `GetEnumerator()`. The object returned by `GetEnumerator()` has a method to move to the next element, and a property that retrieves the current element in the sequence. You will use those two members to enumerate the collection and return the elements. This Interleave method will be an iterator method, so instead of building a collection and returning the collection, you'll use the `yield return` syntax shown above. 
+Naturally, since we split the deck into halves, we'll need to join those halves together. In code, this means we'll be enumerating both of the sequences we acquired through <xref:System.Linq.Enumerable.Take%2A> and <xref:System.Linq.Enumerable.Skip%2A> at once, *interleaving* the elements, and creating one object: our now-shuffled deck of cards. Writing a LINQ method that works with two sequences requires that you understand how <xref:System.Collections.Generic.IEnumerable%601> works.
```

Comment:
I'm definitely ignorant about the usage of sequence instead of object or collection. I'll change this to sequence since I've seen it in other docs pages already. Is sequence more broad/a catch-all of sorts?

Reply:
Good question @jmarkman 

I'd prefer "sequence" over "object" here because I'd usually think of "object" as a single item. The "sequence" in LINQ would mean any `IEnumerable<T>`.